File: Python/majority-element/solution1.py
# Runtime complexity: O(nlogn) because sorting takes nlogn times
# Space complexity: O(1) since we didn't use extra space to store some values

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        return nums[len(nums) // 2]

File: Java/missing-number/Solution1.java
//Time complexity of this code is O(nlogn). Sorting the array takes time complexity of O(nlogn).
//Therefore, the overall complexity is nlogn.

class Solution {
    public int missingNumber(int[] nums) {
        Arrays.sort(nums);
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i) {
                return i;
            }
        }

        return nums.length;
    }
}